sycl/doc/extensions/proposed/sycl_ext_intel_fpga_kernel_interface_properties.asciidoc

```diff
@@ -0,0 +1,198 @@
+or may be in a preliminary state. The specification itself may also change in
+incompatible ways before it is finalized. Shipping software products should not
+rely on APIs defined in this specification.
+
+== Version
+
+Revision: 1
+
+== Contributors
+
+Joe Garvey, Intel +
+Abhishek Tiwari, Intel
+
+== Dependencies
+
+This extension is written against the SYCL 2020 specification, Revision 4 and
+the following extensions:
+
+- link:sycl_ext_oneapi_properties.asciidoc[sycl_ext_oneapi_properties]
+- link:sycl_ext_oneapi_kernel_properties.asciidoc[sycl_ext_oneapi_kernel_properties]
+
+== Feature Test Macro
+
+This extension provides a feature-test macro as described in the core SYCL
+specification section 6.3.3 "Feature test macros".  Therefore, an
+implementation supporting this extension must predefine the macro
+`SYCL_EXT_ONEAPI_KERNEL_PROPERTIES` to one of the values defined in the table
+below.  Applications can test for the existence of this macro to determine if
+the implementation supports this feature, or applications can test the macro's
+value to determine which of the extension's APIs the implementation supports.
+
+[%header,cols="1,5"]
+|===
+|Value |Description
+|1     |Initial extension version.  Base features are supported.
+|===
+
+== Overview
+
+This extension specifies the kernel properties designed for kernels that target
+the FPGA device. These properties are supported by the Intel FPGA backend
+compiler.
+
+== Kernel Properties
+
+```c++
+namespace sycl {
+namespace ext {
+namespace oneapi {
+namespace experimental {
+
+// Corresponds to ip_interface_streaming
+struct ip_interface_streaming_key {
+  template <bool is_stall_free_return>
+  using value_t = property_value<ip_interface_streaming_key, std::bool_constant<bool, is_stall_free_return>>;
+}; // ip_interface_streaming_key
+
+// Corresponds to ip_interface_streaming_stall_free_return
+struct ip_pipelined_key {
+  using value_t = property_value<ip_pipelined_key>;
+}; // ip_pipelined_key
+
+template<bool is_stall_free_return = false>
+inline constexpr ip_interface_streaming_key::value_t<is_stall_free_return> ip_interface_streaming;
+inline constexpr bool ip_stall_free_return = true;
+
+inline constexpr ip_interface_streaming_key::value_t ip_interface_streaming_stall_free_return;
+
+inline constexpr ip_interface_streaming_key::value_t ip_pipelined;
+
+} // namespace experimental
+} // namespace oneapi
+} // namespace ext
+} // namespace sycl
+```
+
+|===
+|Property|Description
+|`ip_interface_streaming`
+|The `ip_interface_streaming` property adds the requirement that the kernel must
+ have dedicated ports for input / output signals in the generated
+ Register Transfer Level (RTL) module.
+ 
+ The property takes a constant boolean template argument. When the value of the
+ argument is `true` it adds the requirement that the kernel module should not
+ take a 'stall' control flow signal from downstream logic as input. The kernel
```

Comment:
Why is the stall-freedom a template parameter of ip_interface_streaming instead of its own property?  How can a user guarantee that done signals are not lost in a pipelined kernel with a CSR interface if there is no way to backpressure the done signal?

Reply:
It should be its own property. Updated.